libqpdf/QPDFObjectHandle.cc

```diff
@@ -3187,12 +3116,15 @@ QPDFObjectHandle::assertPageObject()
     }
 }
 
-void
-QPDFObjectHandle::dereference()
+bool
+QPDFObjectHandle::dereference(bool unguarded)
```

Comment:
Please change `unguarded` to `allow_uninitialized` and get rid of the default for the method. Since it's a private method, I'd rather have an explicit value. It looks like `allow_uninitialized` might have the opposite sense of `unguarded`. To me, `unguarded` doesn't mean anything, but if anything, I would think `unguarded == true` would skip the check rather than perform it.

Reply:
By unguarded I tried to express that is not called in an
```cpp
if (dereference()) {
    ...
}
```
construct. At least some of the existing calls of  the form
```cpp
dereference(true);
```
are actually called with uninitialized QPDFObjectHandles  and do actually throw exception. My preferred solution would  be to change those calls to to be guarded by an if statement and once this is done loose the `unguarded` parameter completely.

Reply:
Or loose the calls completely and replace with a comment as per your comment below.